### PR TITLE
Makes ventcrawl much easier on the client

### PR DESCRIPTION
**Pull request by kriskog**
[Changelogs]: # (Your PR should contain a detailed changelog of notable changes, titled and categorized appropriately. This includes, new features, sprites, sounds, balance changes, admin tools, map edits, removals, big refactors, config changes, hosting changes and important fixes. An example changelog has been provided below for you to edit. If you need additional help, read https://github.com/tgstation/tgstation/wiki/Changelogs)

:cl: Skoglol
fix: Reduced ventcrawl lag greatly.
/:cl:

[why]: # (Please add a short description [two lines down] of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding.)
Basically prevents adding pipe images to client if they are outside view range. I don't know of any situations where the view range while vent crawling is higher than the standard 7, if there are let me know so I can up it.

Partial fix of #41433 as vent crawling is still fairly expensive for the server, but at least alleviates the client lag. See the short videos below for a comparison.

[Before](https://i.imgur.com/9a8d2zw.mp4)
[After](https://i.imgur.com/9J0SWMN.mp4)

### CODE REVIEW

File: code/modules/mob/living/ventcrawling.dm [hunk 1]
```diff
@@ -98,7 +98,9 @@ GLOBAL_LIST_INIT(ventcrawl_machinery, typecacheof(list(
 			A.pipe_vision_img.plane = ABOVE_HUD_PLANE
 		pipes_shown += A.pipe_vision_img
 		if(client)
-			client.images += A.pipe_vision_img
+			if(A.loc.x >= starting_machine.loc.x - 8 && A.loc.x <= starting_machine.loc.x + 8)
+				if(A.loc.y >= starting_machine.loc.y - 8 && A.loc.y <= starting_machine.loc.y + 8)
```
Comment (Cyberboss) [inline code/modules/mob/living/ventcrawling.dm]: Condense all three ifs into one please

**Review by Cyberboss — DISMISSED** [review]

File: code/modules/mob/living/ventcrawling.dm (hunk 1 above)
Comment (kriskog) [reply code/modules/mob/living/ventcrawling.dm]: Done

**Comment by Cyprex** [comment]
Thank you!!!!

**Comment by optimumtact** [comment]
Why not just check the actual client view range and add the pipes?

File: code/modules/mob/living/ventcrawling.dm [hunk 2]
```diff
@@ -97,7 +97,7 @@ GLOBAL_LIST_INIT(ventcrawl_machinery, typecacheof(list(
 			A.pipe_vision_img = image(A, A.loc, layer = ABOVE_HUD_LAYER, dir = A.dir)
 			A.pipe_vision_img.plane = ABOVE_HUD_PLANE
 		pipes_shown += A.pipe_vision_img
-		if(client)
+		if(client && A.loc.x >= starting_machine.loc.x - 8 && A.loc.x <= starting_machine.loc.x + 8 && A.loc.y >= starting_machine.loc.y - 8 && A.loc.y <= starting_machine.loc.y + 8)
```
Comment (ShizCalev) [inline code/modules/mob/living/ventcrawling.dm]:
```suggestion
		if(client && in_view_range(client.mob, A))
```

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/ventcrawling.dm (hunk 2 above)
Comment (ShizCalev) [reply code/modules/mob/living/ventcrawling.dm]: that said, i'm not sure if anything in this for is really necessary if there isn't a client to be showing the pipes to in the first place. could probably nest `for(var/X in totalMembers)` under a `if(client`) check to save some processing to start with.

Comment (kriskog) [reply code/modules/mob/living/ventcrawling.dm]: Oh cool, I was looking for a proc like that. Tested, works well. I also moved the if(client) out of the for loop, which got me thinking. 
We don't really need to check if a pipe has an image set if we never see it, so that is also inside the if now. Not moving pipes_shown the first time was an oversight, as it is used to remove images in the remove_ventcrawl() proc.

**Review by ShizCalev — APPROVED** [review]
